### DISCUSSION TITLE
How to run build from Deploy hook even if branch is defined in Ignored Build Step

### DISCUSSION THREAD

**Discussion by sqpollen** [vercel/community#2764] [Help] [answered]
### Summary

We are running a Next.js / Sanity headless CMS setup.

We have the following config for Ignore Build Step as we are deploying to Vercel from a Bitbucket Pipeline.

`if [[ "$VERCEL_GIT_COMMIT_REF" == "main" ]] ; then   echo "Skip build on main branch";   exit 0; else   exit 1; fi`

This works well

But we want to trigger deployed from Sanity when users publish new content. We have defined the deploy hook for the `main` branch and given this to Sanity. When the user publishes content the deploy hook triggers and the build starts but it is instantly cancelled because of the above Ignore Build Step.

How can we have both a Ignored Build Step AND use the deploy hooks?

Something like this 

`if [[ "$VERCEL_GIT_COMMIT_REF" == "main" && $DEPLOY_HOOK_TRIGGER=false ]] ; then   echo "Skip build on main branch";   exit 0; else   exit 1; fi`

### Steps to Reproduce

<img width="1473" alt="image" src="https://github.com/vercel/community/assets/99855230/618b6089-76e1-4073-aeb9-ca87ab81b1fd">

<img width="1473" alt="image" src="https://github.com/vercel/community/assets/99855230/59c49608-36c2-4536-845e-8a2af828bf40">

**Comment by sqpollen** [comment]
Any ideas anyone?

**Reply by amyegan** [reply, marked as answer]
Have you tried [`git.deploymentEnabled`](https://vercel.com/docs/concepts/projects/project-configuration/git-configuration#git.deploymentenabled) instead of Ignored Build Step to disable automatic deployment from the branch?

**Comment by sqpollen** [comment]
> Have you tried [`git.deploymentEnabled`](https://vercel.com/docs/concepts/projects/project-configuration/git-configuration#git.deploymentenabled) instead of Ignored Build Step to disable automatic deployment from the branch?

Great thank you - that worked!
